src/core/experimental/rcptr.d

```diff
@@ -0,0 +1,208 @@
+// Written in the D programming language.
+/**
+This module provides a shared pointer implementation with memory management
+through reference counting.
+
+License: $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors: Les De Ridder
+
+Source: $(DRUNTIMESRC core/experimental/rcptr.d)
+*/
+module core.experimental.rcptr;
+
+///
+unittest
+{
+    import core.stdc.stdlib : calloc;
+    auto allocInts = (size_t count) => cast(int*) calloc(count, int.sizeof);
+
+    __rcptr!int a; //default ctor
+    assert(a.get is null);
+
+    auto b = __rcptr!int(allocInts(10)); //ptr ctor
+
+    {
+        auto c = a; //copy ctor
+        assert(c.get == a.get);
+        assert(c.get != b.get);
+
+        c = b; //opAssign
+        assert(c.get != a.get);
+        assert(c.get == b.get);
+    }
+
+    assert(a.count is null);
+    assert(*b.count == 0);
+}
+
+/**
+A reference counted shared pointer that can be used to implement reference
+counted data structures.
+
+The reference count is automatically incremented/decremented on assignment,
+copy (construction), and destruction. When there are no more references to the
+pointer, the reference count is automatically deallocated and the referenced
+pointer is `free`d.
+
+Implementation: The internal implementation of `__rcptr` uses `malloc`/`free`.
+*/
+struct __rcptr(T)
+{
+    alias CounterType = int;
+
+    private T* ptr = null;
+    private shared(CounterType)* count = null;
```

Comment:
Still `shared` :(

Reply:
Also not invasive; I will not use this. As the only person I'm aware of who has been harping on an on about RC since 2009, it would be a shame to see someone like me not use this.